packages/flutter_tools/test/commands.shard/hermetic/screenshot_command_test.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:flutter_tools/src/cache.dart';
+import 'package:flutter_tools/src/commands/screenshot.dart';
+import 'package:flutter_tools/src/vmservice.dart';
+
+import '../../src/common.dart';
+import '../../src/context.dart';
+import '../../src/test_flutter_command_runner.dart';
+
+void main() {
+  setUpAll(() {
+    Cache.disableLocking();
+  });
+
+  group('Validate screenshot options', () {
+    testUsingContext('rasterizer and skia screenshots do not require a device', () async {
+      openChannelForTesting = (String url, {CompressionOptions compression, Logger logger}) async {
+        expect(url, 'ws://localhost:8181/ws');
+        throw Exception('dummy');
+      };
+
+      await expectLater(() => createTestCommandRunner(ScreenshotCommand())
+        .run(<String>['screenshot', '--type=skia', '--observatory-uri=http://localhost:8181']),
+        throwsException,
```

Comment:
should we be checking what exception is thrown exactly? it seems like if this regressed, it would still actually throw an exception...

Reply:
Done